### PR TITLE
Improving /setuid

### PR DESCRIPTION
**Pull request by dbemiller**
Updated branch from the suggestion [here](https://github.com/prebid/prebid-server/pull/86#discussion_r134610843) in #86.

This fixes two issues. I grouped them into the same PR because the code overlapped.

- Add metrics for counting calls to /setuid, separated by bidder.
- Re-sync users to audienceNetwork if we were told that their UID was 0.

The point of (2) is to circumvent around some questionable behavior from the Facebook user sync servers. Currently, they redirect to/setuid with a sentinel value of "0" if the user isn't logged into Facebook at the time. This is (at least partially) responsible for some really bad user sync rates, because prebid-server saves those 0s, considers the user to be synced, and then stops trying to sync later.

In reality, 0 is an invalid user ID... but eventually, we'd expect the user to log into Facebook, so that the audience network can sync with their real UID.

### CODE REVIEW

File: pbs/usersync.go [hunk 1]
```diff
@@ -47,30 +61,162 @@ func ParseUIDCookie(r *http.Request) *PBSCookie {
...
+		uids:     make(map[string]string),
+		birthday: timestamp(),
+	}
+}
+
+// PBSCookie is the cookie used in Prebid Server.
+//
+// To get an instance of this from a request, use ParsePBSCookieFromRequest.
+// To write an instance onto a response, use SetCookieOnResponse.
+type PBSCookie struct {
```
Comment (nhedley) [inline pbs/usersync.go]: Any chance you can define this at the top of the file? Think it makes it a bit more readable.

Comment (dbemiller) [inline pbs/usersync.go]: Not sure if you meant "move the method declarations too." I just moved the core struct for now.

Truth be told, I want to move all this code into a `usersync` package and split it into a few files. I didn't want to make a massive diff for this PR though.

Comment (nhedley) [inline pbs/usersync.go]: I only meant the core struct, thanks for moving. Agree that the general refactoring should be done in a separate PR.
